plugins/api/src/replica_close.cpp

```diff
@@ -425,49 +450,41 @@ namespace
                 }
             }
 
-            // Remove the agent's PID from the replica access table.
-            auto entry = is_write_operation
-                ? ix::replica_access_table::erase_pid(l1desc.replica_token, getpid())
-                : std::nullopt;
-
-            // Close the underlying file object.
-            if (const auto ec = close_physical_object(*_comm, l1desc.l3descInx); ec != 0) {
-                if (entry) {
-                    ix::replica_access_table::restore(*entry);
-                }
-
-                log::api::error("Failed to close file object [error_code={}].", ec);
-                update_replica_status(*_comm, l1desc, STALE_REPLICA, false);
-                return ec;
-            }
-
-            const auto ec = free_l1_descriptor(l1desc_index);
-
-            if (ec != 0) {
-                update_replica_status(*_comm, l1desc, STALE_REPLICA, false);
+            // Remove the agent's PID from the replica access table for this replica. The replica access table entry is
+            // only erased when everythng else is successful.
+            if (is_write_operation) {
+                irods::experimental::replica_access_table::erase_pid(l1desc.replica_token, getpid());
             }
```

Comment:
Looks like this can be merged into the block above.

Reply:
Indeed

Reply:
Done